src/org.xtuml.bp.ui.canvas/src/org/xtuml/bp/ui/canvas/Gr_c.java

```diff
@@ -1,927 +1,941 @@
+// External Entity.
+//
+import org.eclipse.jface.preference.JFacePreferences;
+import org.eclipse.jface.resource.FontDescriptor;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.Cursor;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.FontData;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.graphics.TextStyle;
+import org.eclipse.swt.widgets.Canvas;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.ScrollBar;
+
+import org.xtuml.bp.core.CorePlugin;
+import org.xtuml.bp.core.Justification_c;
+import org.xtuml.bp.core.Style_c;
+// Graphics Platform
+public class Gr_c {
+  public static Canvas cur_canvas = null;
+  public static Model_c cur_model = null;
+
+  private static int[] triangle = null;
+  private static int[] openArrow = null;
+  private static int[] cross;
+  private static int lastX = 0;
+  private static int lastY = 0;
+  public static void Setlastx(int value) {
+    Gr_c.lastX = value;
+  }
+  public static void Setlasty(int value) {
+    Gr_c.lastY = value;
+  }
+  public static void Drawtext(final GCDelegate Context, final int Justified_to, final String Text, final int Text_style, final int X, final int Y) {
+    // TODO #4859 there needs to be a case here to handle all of our supported text styles found in
+    // the Style EDT
+	// Current cases should be changed to use the TextLayout/TextStyle ala UNDERLINED below.
+    String[] toPrint = Text.split("\n");  //$NON-NLS-1$
+    int yOffset = 0;
+    if (Justified_to == Justification_c.Center_in_X) {
+      for (int i = 0; i < toPrint.length; i++) {
+        int width = Scale(Context.textExtent(toPrint[i]).x);
+        Context.drawText(toPrint[i], (int) ((X - (width / 2)) * m_ZoomFactor), (int) ((Y + yOffset) * m_ZoomFactor), true);
+        yOffset = yOffset + Scale(Context.textExtent(toPrint[i]).y);
+      }
+    } else if (Justified_to == Justification_c.Center) {
+      // Start by Centering the text block in the y axis
+      yOffset = -Scale(Context.textExtent(Text).y) / 2;
+      for (int i = 0; i < toPrint.length; i++) {
+        int width = Scale(Context.textExtent(toPrint[i]).x);
+        Context.drawText(toPrint[i], (int) ((X - (width / 2)) * m_ZoomFactor), (int) ((Y + yOffset) * m_ZoomFactor), true);
+        yOffset = yOffset + Scale(Context.textExtent(toPrint[i]).y);
+      }
+    } else if (Justified_to == Justification_c.Right) {
+      for (int i = 0; i < toPrint.length; i++) {
+        int width = Scale(Context.textExtent(toPrint[i]).x);
+        Context.drawText(toPrint[i], (int) ((X - width) * m_ZoomFactor), (int) ((Y + yOffset) * m_ZoomFactor), true);
+        yOffset = yOffset + Scale(Context.textExtent(toPrint[i]).y);
+      }
+    } else {
+      if ( Style_c.Underlined == Text_style ) {
```

Comment:
This code also needs to be applied above. With this, underline will only work for left justified text